util/graph.js

```diff
@@ -0,0 +1,77 @@
+/** Class that signifies a graph */
+class Graph {
+    /**
+      * The underlying graph object. This is an object with nodes and edges.
+      * The class contains an adjacency list property that represents the edges
+     * between each of the nodes. Each edge can have attributes such as weight (which 
+     * by default is 10) and whether or not the edge is directed.
+     * 
+      */
+    adjacencyList;
+ 
+    constructor() {
+       this.adjacencyList = []
+    }
+ 
+    clear() {
+       this.adjacencyList = []
+       return true
+    }
+ 
+    addNode(node) {
+       if (!this.adjacencyList[node]) {
+          this.adjacencyList[node] = []
+       }
+    }
+ 
+    addNodes(nodes) {
+       if (!Array.isArray(nodes)) {
+          return false
```

Comment:
https://github.com/tair/mb4-service/pull/58/files#r1170651773
"Instead of returning false, this should throw an error. We want to always make sure that arrays are passed in."